### DISCUSSION TITLE
How are you supposed to migrate an existing API to DateTimeISO?

### DISCUSSION THREAD

**Discussion by capaj** [MichalLytek/type-graphql#1526] [Q&A] [answered]
In the latest version there is a new breaking change:
  > exported GraphQLISODateTime scalar has now a name DateTimeISO
  
  we have many client side queries, which use `DateTime` type.
  How are we supposed to migrate them?
  Ideally I could keep using both types in my schema and slowly replace `DateTime` one by one with `DateTimeISO`, but I don't think that is possible with type-graphql or am I missing something?

**Comment by MichalLytek** [comment]
I don't think in GraphQL we reference scalar name in query document. You mean in your FE code that consumes the returned data and rely on `__typename` check?

**Reply by MichalLytek** [reply]
Ah I see, in the query parameters you type the type 😄 
Doesn't find and replace enough for this change? 

You can always restore the name by re-registering a modified version of this scalar.

**Comment by Manivela** [comment]
adding 
`scalarsMap: [
    {
      type: Date,
      scalar: GraphQLDateTime
    }
 ]` 
to the buildSchema with the import 
`import { GraphQLDateTime } from 'graphql-scalars';`
brings back the old type and you can specify
`@Field(() => GraphQLISODateTime)`
on any Date fields you want to use as DateTimeISO on the frontend.

I don't really understand what's going on here and what's the difference but there you go 😅

**Comment by iamchathu** [comment]
It's breaking change for the Frontend. We need to first release backend without breaking the frontend. Is there way to keep the scalar name to old value?

**Comment by MichalLytek** [comment, marked as answer]
This guide should help you guys! 😉 
https://typegraphql.com/docs/next/migration-guide.html#new-datetimeiso-scalar-name-in-schema

@capaj @iamchathu @Manivela
